core/src/main/java/io/ddf/content/Schema.java

```diff
@@ -1,655 +1,193 @@
-      .synchronizedList(new ArrayList<Column>());
-
-  private DummyCoding dummyCoding;
-
-  /**
-   * Constructor that can take a list of columns in the following format:
-   * "<name> <type>, <name> <type>". For example,
-   * "id string, description string, units integer, unit_price float, total float"
-   * . This string will be parsed into a {@link List} of {@link Column}s.
-   *
-   * @param columns
-   */
-  @Deprecated
-  // Require tableName at all times, even null
-  public Schema(String columns) throws DDFException {
-    this.initialize(null, this.parseColumnList(columns));
-  }
-
-  /**
-   * Constructor that can take a list of columns in the following format:
-   * "<name> <type>, <name> <type>". For example,
-   * "id string, description string, units integer, unit_price float, total float"
-   * .
-   *
-   * @param tableName
-   * @param columns
-   */
-  public Schema(String tableName, String columns) throws DDFException {
-    this.initialize(tableName, this.parseColumnList(columns));
-  }
-
-  @Deprecated
-  // Require tableName at all times, even null
-  public Schema(List<Column> columns) throws DDFException {
-    this.initialize(null, columns);
-  }
-
-  public Schema(String tableName, List<Column> columns) throws DDFException {
-    this.initialize(tableName, columns);
-  }
-
-  @SuppressWarnings("unchecked")
-  public Schema(String tableName, Column[] columns) throws DDFException {
-    this.initialize(tableName, Arrays.asList(columns));
-  }
-
-  private void initialize(String tableName, List<Column> columns) throws DDFException {
-    this.setColumns(columns);
-    this.mTableName = tableName;
-  }
-
-  public static void validateSchema(Schema schema) throws DDFException {
-    if(schema != null && schema.getColumns() != null) {
-      validateColumnNames(schema.getColumnNames());
-    }
-  }
-
-  private static void validateColumnNames(List<String> names) throws DDFException {
-    Set<String> columnSet = new HashSet<String>();
-    if(names != null) {
-      for(String name: names) {
-        if(columnSet.contains(name)) {
-          throw new DDFException(String.format("Duplicated column name %s", name));
-        } else {
-          if(!Utils.isAlphaNumeric(name)) {
-            throw new DDFException(String.format("Invalid column name %s, only allow alphanumeric (uppercase and lowercase a-z, numbers 0-9) " +
-                "and dash (\"-\") and underscore (\"_\")", name));
-          }
-          columnSet.add(name);
-        }
-      }
-    } else {
-      throw new DDFException("names is null");
-    }
-  }
-
-  private List<Column> parseColumnList(String columnList) {
-    if (Strings.isNullOrEmpty(columnList))
-      return null;
-    String[] segments = columnList.split("\\s*,\\s*");
+@NotThreadSafe
+public  class Schema {
 
-    mColumns.clear();
-    for (String segment : segments) {
-      if (Strings.isNullOrEmpty(segment))
-        continue;
+  protected List<String> colNames;
```

Comment:
Why is there are separate list of column names and columns? Why not just one list of columns?